File: av1/numero-primo-java/NumeroPrimo/NumeroPrimo.java
public class NumeroPrimo {
    public static boolean isPrime(int number) {
        if (number <= 1) {
            return false;
        }

        if (number == 2) {
            return true;
        }

        if (number % 2 == 0) {
            return false;
        }

        for (int i = 3; i <= Math.sqrt(number); i += 2) {
            if (number % i == 0) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        int numero = 18;
        if (isPrime(numero)) {
            System.out.println(numero + " é primo.");
        } else {
            System.out.println(numero + " não é primo.");
        }
    }
}

File: av1/numeroPrimo-js/numeroPrimo.js
function isPrime(number) {
    if (number <= 1) {
        return false;
    }

    if (number === 2) {
        return true;
    }

    if (number % 2 === 0) {
        return false;
    }

    for (let i = 3; i <= Math.sqrt(number); i += 2) {
        if (number % i === 0) {
            return false;
        }
    }

    return true;
}

const numero = 48;
if (isPrime(numero)) {
    console.log(`${numero} é primo.`);
} else {
    console.log(`${numero} não é primo.`);
}
